File: backend/schemas/job.py
from pydantic import BaseModel


class JobResponse(BaseModel):
    id: str
    external_id: str
    source: str
    title: str
    company: str
    location: str
    remote_type: str
    salary_min: int | None = None
    salary_max: int | None = None
    description: str
    url: str
    posted_at: str | None = None
    fetched_at: str
    match_score: int | None = None

    class Config:
        from_attributes = True

File: packages/shared/src/types/job.ts
export interface Job {
  id: string;
  external_id: string;
  source: string;
  title: string;
  company: string;
  location: string;
  remote_type: 'remote' | 'hybrid' | 'onsite';
  salary_min?: number;
  salary_max?: number;
  description: string;
  url: string;
  posted_at: string;
  fetched_at: string;
  match_score?: number;
}

export interface JobSearchParams {
  query: string;
  location?: string;
  remote_type?: string;
  salary_min?: number;
  salary_max?: number;
  page?: number;
  limit?: number;
}
